classes/table/reengagement_participants.php

```diff
@@ -290,33 +290,39 @@ public function query_db($pagesize, $useinitialsbar = true) {
         list($twhere, $tparams) = $this->get_sql_where();
         $psearch = new \mod_reengagement\table\reengagement_search($this->course, $this->context, $this->filterset);
 
-        $total = $psearch->get_total_participants_count($twhere, $tparams);
-
-        $this->pagesize($pagesize, $total);
-
         $sort = $this->get_sql_sort();
 
-        $rawdata = $psearch->get_participants($twhere, $tparams, $sort, $this->get_page_start(), $this->get_page_size());
+        $rs = $psearch->get_participants(
+            $twhere,
+            $tparams,
+            $sort,
+            (int)$this->get_page_start(),
+            (int)$this->get_page_size()
+        );
 
         $this->rawdata = [];
-        foreach ($rawdata as $user) {
+        $total = 0;
+
+        foreach ($rs as $user) {
+            if ($total === 0) {
+                $total = (int)$user->fullcount; // <- confirmed by your dump
+            }
             $this->rawdata[$user->id] = $user;
         }
-        $rawdata->close();
+        $rs->close();
```

Comment:
I don't really like the change in name from $rs to $rawdata here... typically $rawdata suggests that we have an array or data object, but the get_participants() function returns a recordset (hence why $rs is used.)

although - interestingly - all existing core uses of the get_participants data don't appear to close the recordset which feels like a core bug to me....

Reply:
ah - hold on - it is rawdata.... ignore that - I'm reading the patch wrong.